csrc/scheduler/matmul_utils.cpp

```diff
@@ -236,31 +236,42 @@ std::string isMatmulFusionDefinitionSupported(
 
   // Fusion topology check
   {
-    auto entry = roles_map.find(MatmulRole::INPUT_A);
+    // We will check that all operands have same dimension
+    int64_t operand_dim = -1;
+
+    // Track TensorViews with assigned roles so we can check that all inputs and
+    // outputs have recognized roles
     std::set<TensorView*> tvs_with_roles;
 
-    if (entry != roles_map.end()) {
-      if (MATMUL_CORE_ROLES_EXPECTED_COUNT == entry->second.size()) {
-        tvs_with_roles.insert(entry->second.begin(), entry->second.end());
+    for (MatmulRole role : {MatmulRole::INPUT_A, MatmulRole::INPUT_B}) {
+      auto entry = roles_map.find(role);
+      if (entry != roles_map.end()) {
+        if (MATMUL_CORE_ROLES_EXPECTED_COUNT == entry->second.size()) {
+          tvs_with_roles.insert(entry->second.begin(), entry->second.end());
+          for (TensorView* tv : entry->second) {
+            if (operand_dim == -1) {
```

Comment:
Just to confirm, we do not support 3D inputs in `LinearOp` then in this PR?

Reply:
That's correct. A LinearOp with 3D input [B M K] and 2D weights [N K] would be expanded like input [B M 1 K] weight [1 1 N K] so you can see that the "batch" dimension in this case is really another M dimension, which we do not support (until #2303).